source/fe/fe_nedelec.cc

```diff
@@ -215,11 +215,362 @@ template <int dim>
+    {{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+      -1, -1, -1, -1, -1, -1, -1, -1, -1, -1}, // 1
+     {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+     {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}},
+    {{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+      -1, -1, -1, -1, -1, -1, -1, -1, -1, -1}, // 2
+     {1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1},
+     {0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1}},
+    {{0,  5,  10, 15, 1,  6,  11, 16, 2,  7,
+      12, 17, 3,  8,  13, 18, 4,  9,  14, 19}, // 3
+     {0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1},
+     {1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1}},
+    {{0,  5,  10, 15, 1,  6,  11, 16, 2,  7,
+      12, 17, 3,  8,  13, 18, 4,  9,  14, 19}, // 4
+     {1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0},
+     {1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0}},
+    {{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+      -1, -1, -1, -1, -1, -1, -1, -1, -1, -1}, // 5
+     {1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0},
+     {1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0}},
+    {{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+      -1, -1, -1, -1, -1, -1, -1, -1, -1, -1}, // 6
+     {0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1},
+     {1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1}},
+    {{0,  5,  10, 15, 1,  6,  11, 16, 2,  7,
+      12, 17, 3,  8,  13, 18, 4,  9,  14, 19}, // 7
+     {1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1},
+     {0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1}}};
+
+  static const int *swap_table_array[5] = {&c_swap_table_0,
+                                           &c_swap_table_1[0][0][0],
+                                           &c_swap_table_2[0][0][0],
+                                           &c_swap_table_3[0][0][0],
+                                           &c_swap_table_4[0][0][0]};
+
+  static const int row_length[5] = {0, 2, 6, 12, 20};
+  static const int table_size[5] = {
+    0, 8 * 3 * 2, 8 * 3 * 6, 8 * 3 * 12, 8 * 3 * 20};
+
+  // Only three-dimensional Nedelec finite elements are treated. The
+  // two-dimensional Nedelec finite elements only need sign adjustments of the
+  // line dofs. These adjustments are done by
+  // internal::FE_PolyTensor::get_dof_sign_change_nedelec(...)
+  // in fe_poly_tensor.cc. The notions of curl and curl-conforming finite
+  // elements in higher-dimensional spaces, dim >3, are somewhat unclear as
+  // curl, strictly peaking, exists only in the three-dimensional space.
+  if (dim != 3)
     return;
 
-  // TODO: Implement this for this class
+  const unsigned int k = this->tensor_degree() - 1;
+
+  // The Nedelec finite elements of the lowermost order have no quad dofs.
+  if (k == 0)
+    return;
+
+  // The finite element orders > 4 are not implemented.
+  if (k > 4)
```

Comment:
Can you change this into
``AssertThrow(k<=4, ExcNotImplemented());``

Reply:
Agree. I will correct this.